### PR TITLE
Getting of routes now in a function 

### PR DESCRIPTION
**Pull request by skcoulter**
This is in preparation for another patch.

Signed-off-by: Susan K. Coulter <markus@wf-fe1.lanl.gov>

### CODE REVIEW

File: src/ibtracert.c [hunk 1]
```diff
@@ -778,15 +778,84 @@ static int process_opt(void *context, int ch, char *optarg)
...
 
-int main(int argc, char **argv)
-{
-	int mgmt_classes[3] =
-	    { IB_SMI_CLASS, IB_SMI_DIRECT_CLASS, IB_SA_CLASS };
+static int get_route(char *srcid, char *dstid) {
 	ib_portid_t my_portid = { 0 };
 	ib_portid_t src_portid = { 0 };
 	ib_portid_t dest_portid = { 0 };
 	Node *endnode;
```
Comment (hnrose) [inline src/ibtracert.c:792]: These variables are no longer needed and can now be removed in main(), right ?

File: src/ibtracert.c [hunk 2]
```diff
@@ -827,53 +896,8 @@ int main(int argc, char **argv)
...
-
-	if (!(target_portguid = find_target_portguid(&dest_portid)))
-		IBEXIT("can't reach target lid %d", dest_portid.lid);
-
-	if (!(endnode = find_mcpath(&src_portid, mlid)))
-		IBEXIT("can't find a multicast route from src to dest");
-
-	/* dump multicast path */
-	dump_mcpath(endnode, dumplevel);
+	if (get_route(argv[0],argv[1]) != 0)
```
Comment (hnrose) [inline src/ibtracert.c]: Nit: space between arguments:
get_route(argv[0], argv[1])

**Review by hnrose — CHANGES_REQUESTED** [review]
Title should be something like:

ibtracert.c: Made getting of routes into internal function

in preparation for ...

File: src/ibtracert.c (hunk 1 above)
Comment (skcoulter) [reply src/ibtracert.c:792]: They only reside in the function now, and are no longer in main.
The way git created the patch was kind of confusing. (IMO)

Not sure how to change the title...

Comment (hnrose) [reply src/ibtracert.c:792]: Oh, you're right. Of course, they're not part of the new main.

As far as updating the commit, you can amend it to update the description only and then push -f.

Comment (hnrose) [reply src/ibtracert.c:792]: git rebase -I starting at commit ID prior to this
then amend the first commit
then git push -f

**Comment by weiny2** [comment]
Closing this in favor of the series Hal just finished up.

**Pull request closed without merging**